bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -917,12 +922,26 @@ void BCStateTran::startCollectingStateInternal() {
   startCollectingStateImpl();
 }
 
-void BCStateTran::startCollectingStateImpl() {
+void BCStateTran::setExternalCollectingState(bool newVal) {
+  if (newVal == externalCollectingState_) {
+    LOG_WARN(logger_, "Trying to set into an already existing state:" << KVLOG(newVal, externalCollectingState_));
+    return;
+  }
+  externalCollectingState_ = newVal;
+  LOG_INFO(logger_, std::boolalpha << KVLOG(externalCollectingState_));
+}
+
+void BCStateTran::startCollectingStateExternal() {
   ConcordAssert(running_);
-  if (psd_->getIsFetchingState()) {
+  if (isCollectingState()) {
     LOG_WARN(logger_, "Already in State Transfer, ignore call...");
```

Comment:
What flows should cause multiple state transfer instances?
If there are none, shouldn't this throw?